jaxopt/_src/anderson.py

```diff
@@ -0,0 +1,284 @@
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Anderson Acceleration for fixed point in JAX."""
+
+from typing import Any
+from typing import Callable
+from typing import NamedTuple
+
+from dataclasses import dataclass
+
+import jax.numpy as jnp
+from jax.tree_util import tree_leaves, tree_structure
+
+from jaxopt._src import base
+from jaxopt._src import linear_solve
+from jaxopt._src.tree_util import tree_collapse, tree_uncollapse, tree_l2_norm
+
+
+class AndersonState(NamedTuple):
+  """Named tuple containing state information.
+  
+  Attributes:
+    iter_num: iteration number
+    value: pytree of current estimate of fixed point
+    error: residuals of current estimate
+    v_history: history of previous iterates
+    f_history: fixed_point_fun(v_history)
+  """
+  iter_num: int
+  value: Any
+  error: float
+  v_history: Any
+  f_history: Any
+
+
+@dataclass
+class AndersonAcceleration(base.IterativeSolver):
+  """Multi-dimensional fixed point solver using Anderson.
+
+  Attributes:
+    fixed_point_fun: a function ``fixed_point_fun(x, *args, **kwargs)``
+                     returning a pytree with the same structure and type as x
+                     each leaf must be an array (not a scalar)
```

Comment:
When you need a line return, use 2-space indent as indicated by @shoyer 
```
fixed_point_fun: a function ``fixed_point_fun(x, *args, **kwargs)``
  returning a pytree with the same structure and type as x
  each leaf must be an array (not a scalar)
```
Beyond following the rules, this will ensure that the docstring renders correctly.